third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -219,6 +239,20 @@ void dds_device::impl::create_notifications_reader()
     _notifications_reader->run( rqos );
 }
 
+void dds_device::impl::create_metadata_reader()
+{
+    if( _metadata_reader )
+        return;
```

Comment:
`// We can be called multiple times, once per stream`

Reply:
Yes, the first metadata enabled stream will create the reader for the device. Subsequent calls will just return.

Reply:
Yes meaning you added the comment?

Reply:
Added now :-)